src/webgpu/api/validation/buffer/mapping.spec.ts

```diff
@@ -43,13 +43,17 @@ class F extends ValidationTest {
     if (success) {
       const p = buffer.mapAsync(mode, offset, size);
       await p;
-      this.shouldResolve(p);
     } else {
-      this.expectValidationError(async () => {
-        const p = buffer.mapAsync(mode, offset, size);
-        await p;
-        this.shouldReject(rejectName!, p);
+      let p: Promise<void>;
+      this.expectValidationError(() => {
+        p = buffer.mapAsync(mode, offset, size);
```

Comment:
Hm, maybe I don't understand the problem - Why can't we return `p` instead of assigning it in this way?

```typescript
const p = this.expectValidationError(() => {
  return buffer.mapAsync(..)
});
```

Or, shouldn't the async version work here too?
```typescript
const p = this.expectValidationError(async () => {
  return await buffer.mapAsync(..)
});
```

Or to make it more explicit we could do

```typescript
const { callbackResult: p } = this.expectValidationError(async () => ...);
```

or... if we need the promise after popErrroScope too?

```typescript
const { callbackResult: p, validationPromise } = this.expectValidationError(async () => ...);
```

Reply:
We can, but I realized that passing an async function into expectValidationError is confusing. Either we:
- make expectValidationError async and await on fn(): Causes an async split between pushErrorScope and popErrorScope, so if you forget to await expectValidationError (or choose not to because you don't care about the result), then other code will end up getting caught by the error scope
- make expectValidationError NOT await fn(), but just execute its first microtask and return the resulting promise: Confusing because you might think that the error scope would include the whole function and not just the microtask

Reply:
Right, okay that makes sense because we want it to be very explicit what's in-between the two synchronous push/pop calls. Sounds good to me then, let's document the reasoning in a comment.